### PR TITLE
[field] Fix bug introduced by RTL in single input range fields

### PR DESCRIPTION
**Pull request by alexfauquette**
TODO:

- [x] Add test
- [x] Verify keyboard interaction in RTL display

### CODE REVIEW

File: packages/x-date-pickers-pro/src/internal/models/fields.ts [hunk 1]
```diff
@@ -3,7 +3,16 @@ import { SlotComponentProps } from '@mui/base/utils';
 import TextField, { TextFieldProps } from '@mui/material/TextField';
 import Stack, { StackProps } from '@mui/material/Stack';
 import Typography, { TypographyProps } from '@mui/material/Typography';
-import { BaseFieldProps } from '@mui/x-date-pickers/internals';
+import { BaseFieldProps, FieldSection } from '@mui/x-date-pickers/internals';
+
+export interface RangeFieldSection extends FieldSection {
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/internal/models/fields.ts:8]: You can probably drop `DateRangeFieldSection` in favor of this `RangeFieldSection`

File: packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx:1]: The test files are usually named `{topic}.{componentName}.test.tsx`
It is the nomenclature on the data grid and on the new pickers test files.

I would name this one `selection.SingleInputDateRangeField.test.tsx`

File: packages/x-date-pickers-pro/src/internal/utils/valueManagers.ts [hunk 3]
```diff
@@ -75,34 +76,43 @@ export const rangeFieldValueManager: FieldValueManager<
...
 
-    const sectionsOfStartDate = rawSectionsOfStartDate.map((section, sectionIndex) => {
-      if (sectionIndex === rawSectionsOfStartDate.length - 1) {
+    const sectionsOfStartDate: RangeFieldSectionWithoutPosition[] = rawSectionsOfStartDate.map(
+      (section, sectionIndex) => {
+        if (sectionIndex === rawSectionsOfStartDate.length - 1) {
+          return {
+            ...section,
+            dateName: 'start',
+            separator: `\u2069${' – '}\u2066`,
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/internal/utils/valueManagers.ts:85]: Why do we need to use template string here ?
Is there a difference with this ?

```suggestion
            separator: '\u2069 – \u2066',
```

File: packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx (hunk 2 above)
Comment (alexfauquette) [reply packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx:1]: For now, all the `x-pickers-pro` are like that. I will rename all of them in a follow up PR

File: packages/x-date-pickers-pro/src/internal/utils/valueManagers.ts (hunk 3 above)
Comment (alexfauquette) [reply packages/x-date-pickers-pro/src/internal/utils/valueManagers.ts:85]: Just the result of iterating on different solutions 🙈

File: packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/SingleInputDateRangeField.keyboard.test.tsx:1]: @LukasTy is creating new test files using the correct nomenclature in #7135
Can be done in a follow up PR of course

**Review by flaviendelangle — APPROVED** [review]
Great catch !
We definitely need more tests on the single input range fields, as they are not being played with manually a lot, it's easy to break something and see it weeks later

**Pull request merged**